daemon/containerd/image_prune.go

```diff
@@ -148,5 +159,69 @@ func (i *ImageService) pruneUnused(ctx context.Context, filterFunc imageFilterFu
 			}
 		}
 	}
+
+	// Workaround for https://github.com/moby/buildkit/issues/3797
+	if err := i.unleaseSnapshotsFromDeletedConfigs(ctx, possiblyDeletedConfigs); err != nil {
```

Comment:
Maybe it would be safer to do this in a `defer` on a fresh context(can still have a timeout). If the context gets canceled in the middle of the request then atm either this is not reached and some images remain half deleted or only half of `unleaseSnapshotsFromDeletedConfigs` completes.

If it complicates a lot then maybe not worth it as it should only be a temp workaround for testing experimental feature.

Reply:
Yeah, good point, done.